src/libAtomVM/nifs.c

```diff
@@ -3505,6 +3557,374 @@ static term nif_atomvm_read_priv(Context *ctx, int argc, term argv[])
+    term flags = argv[1];
+    VALIDATE_VALUE(flags, term_is_list);
+    int posix_flags = O_NONBLOCK; // Always be non-blocking
+    while (term_is_nonempty_list(flags)) {
+        term flag = term_get_list_head(flags);
+        VALIDATE_VALUE(flag, term_is_atom);
+        // SUSv1 flags
+        if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_RDONLY_ATOM)) {
+            posix_flags |= O_RDONLY;
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_WRONLY_ATOM)) {
+            posix_flags |= O_WRONLY;
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_RDWR_ATOM)) {
+            posix_flags |= O_RDWR;
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_APPEND_ATOM)) {
+            posix_flags |= O_APPEND;
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_CREAT_ATOM)) {
+            posix_flags |= O_CREAT;
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_EXCL_ATOM)) {
+            posix_flags |= O_EXCL;
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_NOCTTY_ATOM)) {
+            posix_flags |= O_NOCTTY;
+        // O_NONBLOCK is not optional
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_SYNC_ATOM)) {
+            posix_flags |= O_SYNC;
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_TRUNC_ATOM)) {
+            posix_flags |= O_TRUNC;
+        // SUSv4/2018 edition flags
+#if HAVE_O_EXEC
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_EXEC_ATOM)) {
+            posix_flags |= O_EXEC;
+#endif
+#if HAVE_O_SEARCH
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_SEARCH_ATOM)) {
+            posix_flags |= O_SEARCH;
+#endif
+#if HAVE_O_CLOEXEC
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_CLOEXEC_ATOM)) {
+            posix_flags |= O_CLOEXEC;
+#endif
+#if HAVE_O_DIRECTORY
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_DIRECTORY_ATOM)) {
+            posix_flags |= O_DIRECTORY;
+#endif
+#if HAVE_O_DSYNC
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_DSYNC_ATOM)) {
+            posix_flags |= O_DSYNC;
+#endif
+#if HAVE_O_NOFOLLOW
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_NOFOLLOW_ATOM)) {
+            posix_flags |= O_NOFOLLOW;
+#endif
+#if HAVE_O_RSYNC
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_RSYNC_ATOM)) {
+            posix_flags |= O_RSYNC;
+#endif
+#if HAVE_O_TTY_INIT
+        } else if (globalcontext_is_term_equal_to_atom_string(glb, flag, O_TTY_INIT_ATOM)) {
+            posix_flags |= O_TTY_INIT;
+#endif
+        } else {
+            RAISE_ERROR(BADARG_ATOM);
+        }
+        flags = term_get_list_tail(flags);
+    }
+
+    int ok;
+    const char *path = interop_term_to_string(path_term, &ok);
```

Comment:
I don't think there is any need to declare this as `const char *` since we own this memory and we can do whatever we want.